### ISSUE TITLE
Local of unconstrained type parameter type is reported as NotAnnotated when using var

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#46236]
The local returned from `SemanticModel.GetDeclaredSymbol()` for `y1` has `Type.NullableAnnotation == NullableAnnotation.NotAnnotated`, but should be `NullableAnnotation.Annotated` instead.

Compare with the `GetDeclaredSymbol()` result for `y2` which has `Type.NullableAnnotation == NullableAnnotation.Annotated`.

```C#
#nullable enable

class Program
{
    static T F1<T>(T x1)
    {
        var y1 = x1;
        y1 = default(T);
        return y1;
    }

    static T F2<T>(T x2) where T : class
    {
        var y2 = x2;
        y2 = default(T);
        return y2;
    }
}
```

**Comment by RikkiGibson** [comment]
@333fred is this by design, since we tend to "lie" about the nullable annotation to give a consistent answer to what we give for flow state? e.g. we always give (NotAnnotated, NotNull) or (Annotated, MaybeNull) or "oblivious"?

**Comment by 333fred** [comment]
It might have been by design in C# 8, but with `T?` in C# 9 it no longer feels like that's the case.

**Comment by RikkiGibson** [comment]
The initial state of parameters, though, is MaybeNull for both unconstrained `T` and `T?`, and that is where the `NullableAnnotation.Annotated` is coming from in this public API, right?

**Comment by 333fred** [comment]
I don't believe we're talking about parameters here. We're talking about `var`. And `var` should be annotated, with whatever flow state it gets from the initializer.

**Comment by RikkiGibson** [comment]
Follow-up to offline discussion: this API is supposed to tell the truth 😉 so this is a bug. Thanks for helping me understand the scenario.

**Comment by jcouv** [comment]
Fixed and verified for 17.0 (p1 or p2) by https://github.com/dotnet/roslyn/pull/53691

**Comment by jcouv** [comment]
Also backported fix to 16.11
